packages/react-sdk/src/useSendMessage.ts

```diff
@@ -0,0 +1,18 @@
+'use client'
+
+import { useCallback, useState } from 'react'
+import { useCurrentChannel } from './useChannel'
+
+export const useSendMessage = () => {
+    const channel = useCurrentChannel()
+    const [isSending, setSending] = useState(false)
+    const send = useCallback(
+        async (message: string) => {
+            setSending(true)
+            return channel.sendMessage(message).finally(() => setSending(false))
+        },
```

Comment:
In the same way we wrap the observables, we could probably have one hook that wraps all function calls  we will need to add error handling to everything,

Reply:
100% - nice catch!